### PR TITLE
Add export options in `datum merge`

### LINKED ISSUE #542: Images missing after merging two datasets
Hello all,

I am trying to merge two datasets using the merge command, but unfortunately this is not successful. 

The following is present:
- Two datasets -> coco_instances
- Both have a different number of images
- In the categories part (anotation file) the names are identical
- Folder structure is the same for both (project can import the datasets without any problems).

Now comes my problem:
With the merge command I try to merge both datasets.

date merge path/to/dataset1:coco_instances path/to/dataset2:coco_instances

I get a folder with the annotation file, a report and a folder for the images. However, this does not contain the images. But in the annotation file there is a path for every single image. I would like to have the images in the folder and remove the paths in the annotation file. How can I do this?

I cannot import the merged dataset (created folder after merge command) into any empty datum project. 

Datumaro version is 0.2

### PR DESCRIPTION
**Pull request by zhiltsov-max**
### Summary

Closes #542

- Added export options in `datum merge`
- Added a reference to the `patch` command in `merge` docs

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/cli/commands/merge.py [hunk 1]
```diff
@@ -26,13 +28,17 @@
...
-        have few annotations and wish to merge them,
-        taking into consideration potential overlaps and conflicts.
-        This command can try to find a common ground by voting or
+        Merges multiple datasets into one and produces a new dataset.
+        The command can be useful if you have few annotations and wish
+        to merge them, taking into consideration potential overlaps and
+        conflicts. This command can try to find common ground by voting or
         return a list of conflicts.|n
         |n
+        In simple cases, when dataset images are not intersected and new
```
Comment (IRDonch) [inline datumaro/cli/commands/merge.py:37]:
```suggestion
        In simple cases, when dataset images do not intersect and new
```

File: datumaro/cli/commands/merge.py [hunk 2]
```diff
@@ -56,6 +62,13 @@ def build_parser(parser_ctor=argparse.ArgumentParser):
         It can be useful for dataset paths in targets. When not specified,
         the current project's working tree is used.|n
         |n
+        The output format can be specified with the '-f/--format' option.
+        Each dataset format has its own export
+        options, which are passed after the '--' separator (see examples),
+        pass '-- -h' for more info. If not stated otherwise, by default
+        only annotations are exported, to include images pass
```
Comment (IRDonch) [inline datumaro/cli/commands/merge.py:69]:
```suggestion
        only annotations are exported; to include images, pass
```

File: tests/cli/test_merge.py [hunk 3]
```diff
@@ -112,8 +112,8 @@ def test_can_run_multimerge(self):
             dataset2.export(dataset2_url, 'voc', save_images=True)
 
             result_dir = osp.join(test_dir, 'cmp_result')
-            run(self, 'merge', dataset2_url + ':voc', dataset1_url + ':coco',
-                '-o', result_dir)
+            run(self, 'merge', '-o', result_dir,
+                dataset2_url + ':voc', dataset1_url + ':coco')
```
Comment (IRDonch) [inline tests/cli/test_merge.py:116]: How about a test for the new option?

**Review by IRDonch — CHANGES_REQUESTED** [review]

File: tests/cli/test_merge.py (hunk 3 above)
Comment (zhiltsov-max) [reply tests/cli/test_merge.py:116]: Added

**Review by IRDonch — APPROVED** [review]

**Pull request merged**